lib/datadog/appsec/event.rb

```diff
@@ -110,12 +114,30 @@ def self.build_service_entry_tags(event_group)
           tags['_dd.appsec.triggers'] += waf_result.events
 
           waf_result.derivatives.each do |key, value|
-            tags[key] = JSON.dump(value)
+            data = Base64.encode64(gzip(JSON.dump(value)))
+
+            if data.size >= MAX_ENCODED_SCHEMA_SIZE
+              Datadog.logger.debug do
+                "Schema key: #{key} exceed max size value. We do not include it as part of the span tags"
```

Comment:
Just a minor wording suggestion

```suggestion
                "Schema key: #{key} exceeds the max size value. It will not be included as part of the span tags"
```